### PR TITLE
build(actix): Update actix-web to v4.

### PR DESCRIPTION
**Pull request by fourbytes**
Updated sentry-actix to support actix-web v4 (currently still in beta).

### CODE REVIEW

**Review by aramperes — COMMENTED** [review]
Note: this raises the minimum supported Rust version from 1.45 to 1.46: https://github.com/actix/actix-web/pull/1858

The `Test MSRV` job will need to be updated:

https://github.com/getsentry/sentry-rust/blob/e8d0e1b5723dab23f74c2e9cd734859ae36f546f/.github/workflows/ci.yml#L102

**Review by Swatinem — COMMENTED** [review]
lgtm! lints are still failing, and I will wait with merging this until we have a stable release.

File: sentry-actix/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ edition = "2018"
 
 [dependencies]
 sentry-core = { version = "0.22.0", path = "../sentry-core", default-features = false }
```
Comment (JakubKoralewski) [inline sentry-actix/Cargo.toml:15]: I need to add this for my project using actix-web v4 and this sentry-actix branch to compile for some reason:

```suggested
sentry-core = { version = "0.22.0", path = "../sentry-core", default-features = false, features=["client"] }
```

Comment (mlodato517) [reply sentry-actix/Cargo.toml:15]: I also had to do this in my local fork

**Review by mlodato517 — COMMENTED** [review]
I think this needs to update the tests to remove the `&mut`s in [`call_service`](https://github.com/getsentry/sentry-rust/blob/d798b3c594bc05a4ac66beb724ed702dcc6ca7a8/sentry-actix/src/lib.rs#L360) which now takes a [shared reference](https://docs.rs/actix-web/4.0.0-beta.8/actix_web/test/fn.call_service.html).

File: sentry-actix/Cargo.toml (hunk 1 above)
Comment (fourbytes) [reply sentry-actix/Cargo.toml]: Hmm, not sure about this as it seems to be working in my project and wasn't needed in previous versions.

**Comment by fourbytes** [comment]
Latest update should fix those lints, but happy to wait for the full release of v4 for this to be merged.

File: sentry-actix/Cargo.toml (hunk 1 above)
Comment (fourbytes) [reply sentry-actix/Cargo.toml]: Ah, just managed to reproduce this. Honestly not sure why it wasn't happening before. It does appear it is now necessary though so I've updated the PR.

**Comment by ppamorim** [comment]
@fourbytes Any update on this, I can't use it because I am getting this error:
```
the trait `Transform<actix_web::app_service::AppRouting, ServiceRequest>` is not implemented for `Sentry`
```
This error is similar to the one I found here: https://github.com/actix/actix-web/issues/2633

**Comment by Swatinem** [comment]
superceded by #437

**Pull request closed without merging**